lib/modules/versioning/composer/index.ts

```diff
@@ -67,28 +67,33 @@ function normalizeVersion(input: string): string {
 }
 
 function composer2npm(input: string): string {
-  const cleanInput = normalizeVersion(input);
-  if (npm.isVersion(cleanInput)) {
-    return cleanInput;
-  }
-  if (npm.isVersion(padZeroes(cleanInput))) {
-    return padZeroes(cleanInput);
-  }
-  const [versionId, stability] = getVersionParts(cleanInput);
-  let output = versionId;
-
-  // ~4 to ^4 and ~4.1 to ^4.1
-  output = output.replace(
-    regEx(/(?:^|\s)~([1-9][0-9]*(?:\.[0-9]*)?)(?: |$)/g),
-    '^$1'
-  );
-  // ~0.4 to >=0.4 <1
-  output = output.replace(
-    regEx(/(?:^|\s)~(0\.[1-9][0-9]*)(?: |$)/g),
-    '>=$1 <1'
-  );
-
-  return output + stability;
+  return input
```

Comment:
Is this change expected?

Reply:
Yeah. When trying to run it against https://github.com/Lctrs/playground, it failed for some PHP libs that defined constraints like `^7.4|^8`. I had to adapt this method to support normalizing range constraints.

Reply:
can we extract that change to a separate PR for main?

Reply:
Done https://github.com/renovatebot/renovate/pull/20558